### ISSUE TITLE
Feat: Implementação da Arquitetura EfficientNet

### ISSUE THREAD

**Issue by Horusprg** [Horusprg/OACE-Validation#20]
**Prioridade:** Média

**Descrição:**

Crie o módulo PyTorch para a arquitetura **EfficientNet**. Esta arquitetura é conhecida por seu método de *scaling* composto, que otimiza profundidade, largura e resolução de forma eficiente.

#### 🏗️ **Tarefas:**

* Crie o diretório `models/efficientnet/`.
* Crie o arquivo `models/efficientnet/efficientnet_architecture.py`.
* Implemente as classes para os **blocos MBConv (Mobile Inverted Bottleneck Convolution)**, que são a base da EfficientNet.
* Implemente a classe `EfficientNet(nn.Module)` que empilha esses blocos. Ela deve aceitar parâmetros para definir a estrutura da rede, com foco no princípio de *scaling* composto:
    * `width_multiplier`, `depth_multiplier`: Fatores que definem a largura e a profundidade da rede base.
    * `resolution`: Resolução da imagem de entrada (influenciada pelo *scaling*).
    * `num_classes`: Número de classes de saída.
    * Opções para **`dropout_rate`** e **inicialização de pesos** (`weight_init_fn`).
* Implemente a função `generate_efficientnet_architecture`. Essa função deve gerar uma instância da `EfficientNet` baseada em parâmetros que permitem a randomização (ou seleção programática) de:
    * O **modelo base da EfficientNet** (ex: B0, B1, etc., ou diretamente os multiplicadores de largura e profundidade).
    * Configurações de dropout.
* Inclua um bloco `if __name__ == "__main__":` no arquivo para demonstrar a geração de um modelo EfficientNet e um *forward pass* básico.

**Comment by LyanhVini** [comment]
feito!
